CompanyEmployees.Infrastructure.Persistence/Repositories/AttendanceRepository.cs

```diff
@@ -0,0 +1,41 @@
+using CompanyEmployees.Core.Domain.Entities;
+using CompanyEmployees.Core.Domain.Repositories;
+using Microsoft.EntityFrameworkCore;
+
+namespace CompanyEmployees.Infrastructure.Persistence.Repositories;
+
+internal sealed class AttendanceRepository : RepositoryBase<Attendance>, IAttendanceRepository
+{
+    public AttendanceRepository(RepositoryContext repositoryContext)
+        : base(repositoryContext)
+    {
+    }
+
+    public async Task<IEnumerable<Attendance>> GetAllAttendancesAsync(bool trackChanges, CancellationToken ct = default) =>
+        await FindAll(trackChanges)
+            .OrderByDescending(a => a.CheckInTime)
+            .ToListAsync(ct);
+
+    public async Task<Attendance> GetAttendanceAsync(Guid attendanceId, bool trackChanges, CancellationToken ct = default) =>
```

Comment:
Consider adding a nullability annotation (`?`) to the return type of `GetAttendanceAsync` if a matching record might not be found and `SingleOrDefaultAsync` returns null. This helps improve null safety and clarity.

**Suggested change**
```suggestion
public async Task<Attendance?> GetAttendanceAsync(Guid attendanceId, bool trackChanges, CancellationToken ct = default) =>
```